python/mxnet/module/base_module.py

```diff
@@ -244,7 +247,7 @@ def predict(self, eval_data, num_batch=None, merge_batches=True, reset=True,
                 break
             self.forward(eval_batch, is_train=False)
             pad = eval_batch.pad
-            outputs = [out[0:out.shape[0]-pad].asnumpy() for out in self.get_outputs()]
+            outputs = [out[0:out.shape[0]-pad] for out in self.get_outputs()]
```

Comment:
I agree with this change but this is not backward compatible. Maybe add a option to output ndarray and set false as default?